Source/src/modules/ModuleCamera.cpp

```diff
@@ -160,14 +160,14 @@ void Hachiko::ModuleCamera::FocusOnModel(const float3& target, float distance) c
 
 void Hachiko::ModuleCamera::Rotate(float motion_x, float motion_y) const
 {
-    auto* transform = main_camera->GetGameObject()->GetComponent<ComponentTransform>();
+    auto* transform = main_camera_game_object->GetTransform();
```

Comment:
```suggestion
    ComponentTransform* transform = main_camera_game_object->GetTransform();
```

Reply:
Hey, thanks for the review, I marked the ones with autos as resolved since these changes were out of scope of this pr. You are right and, I don't like the use of auto either. But, as it's not my code, I tried not to touch other classes as much as possible to not get lost, and lose the main objective of reworking the transform.